doc/tutorials/nires_howto.rst

```diff
@@ -0,0 +1,396 @@
+observed two offset positions for both the science object and the standard star.
+The numbers in the ``comb_id`` and ``bkg_id`` columns (in the corrected
+:ref:`pypeit_file`) indicate that we create both A-B and B-A images for both the
+standard and science frames.
+
+----
+
+Core Processing
+===============
+
+To perform the core processing of the NIRES data, use :ref:`run-pypeit`:
+
+.. code-block:: bash
+
+    run_pypeit keck_nires_A.pypeit
+
+The code will run uninterrupted until the basic data-reduction procedures
+(wavelength calibration, field flattening, object finding, sky subtraction, and
+spectral extraction) are complete; see :doc:`../running`.  Processing of this
+example dataset takes roughly 10 minutes (on a 2020 MacBook Pro with 16 GB of
+RAM and a 2GHz i5 processor).
+
+As the code processes your data, it will produce a number of files and QA plots
+for you to inspect:
+
+Order Edges
+-----------
+
+The code first uses the ``trace`` frames to find the order edges.  NIRES is a
+fixed-format echelle, meaning that the trace results should always look the
+same.  To show the results of the trace, run, e.g.:
+
+.. code-block:: bash
+    
+    pypeit_chk_edges Masters/MasterEdges_A_7_DET01.fits.gz
+
+which will show the image and overlay the traces (green is the left edge;
+magenta is the right edge); this should open a `ginga`_ window for you if one
+is not open already.  Here is the result from this example dataset:
+
+.. image:: ../figures/nires_trace.png
+   :scale: 60%
+
+An important check is to ensure that the code has correctly traced the bluest
+(left-most) order.  PypeIt currently expects to find all 5 orders and will fault
+if it does not.  
+    
+.. tip::
+
+    If PypeIt faults because it did not find all 5 orders, try adjusting the
+    ``edge_thresh`` parameter; see the :ref:`parameters` and specifically the
+    :ref:`edgetracepar`.
+
+Wavelength Calibration
+----------------------
+
+Next the code performs the wavelength calibration.  Via the :ref:`pypeit_file`,
+we designated two sets of wavelength calibration frames, one for the standard
+star and one for the science frame.  You should inspect the results for both.
+
+First, it's important to understand PypeIt's :ref:`master-naming` convention,
+specifically the calibration group bit identities used in the output file names.
+In this example, two :ref:`masterarc` files are produced:
+``Masters/MasterArc_A_2_DET01.fits`` and ``Masters/MasterArc_A_4_DET01.fits``.
+The ``2`` and ``4`` are the bits associated with the calibration group and link
+back to which files are associated with each frame type.  The
+:ref:`calibrations-calibfile` provides the direct association of input frame
+with output calibration file.
+
+Combined Arc Frame
+++++++++++++++++++
+
+You can view the combined arc frame used for, e.g., the standard star
+observations with `ginga`_:
+
+.. code-block:: bash
+
+    ginga Masters/MastersArc_A_2_DET01.fits
+
+1D Wavelength Solution
```

Comment:
Can we add a warning that a standard failure mode is to use an observation so short that the OH sky lines are not well detected?

Reply:
See at the added warning.